build/pause/windows/pause.c

```diff
@@ -0,0 +1,49 @@
+/*
```

Comment:
I'm curious about the mix of C and go here, is there some background on this image?
I feel these binaries could use a little documentation, I wouldn't block the PR on it, but I think that would be helpful in the future.

pause.c seems pretty self explanatory, but I'm not sure why wincat even exists 🙃

Reply:
Hm, Linux has a ``pause.c`` too, so we're not doing anything new here either. https://github.com/kubernetes/kubernetes/blob/master/build/pause/pause.c

As for wincat, I've explained it on this PR's general comment section.

Reply:
Yes, but in that case pause.c was the only source code for the image. Now that we have 3 sources in 2 languages, I think we should document them at least a little. As I said it could be a follow up, but I want to mention it here.